ShipMonitor/ConfigurationWindow.xaml.cs

```diff
@@ -144,26 +141,19 @@ private void shipsUpdated(object sender, SelectionChangedEventArgs e)
                 shipMonitor()?.Save();
             }
         }
-    }
-
-    public class ValidIpaRule : ValidationRule
-    {
-        private static readonly Regex ipaRegex = new Regex(@"^[bdfɡhjklmnprstvwzxaɪ˜iu\.ᵻᵿɑɐɒæɓʙβɔɕçɗɖðʤəɘɚɛɜɝɞɟʄɡ(ɠɢʛɦɧħɥʜɨɪʝɭɬɫɮʟɱɯɰŋɳɲɴøɵɸθœɶʘɹɺɾɻʀʁɽʂʃʈʧʉʊʋⱱʌɣɤʍχʎʏʑʐʒʔʡʕʢǀǁǂǃˈˌːˑʼʴʰʱʲʷˠˤ˞n̥d̥ŋ̊b̤a̤t̪d̪s̬t̬b̰a̰t̺d̺t̼d̼t̻d̻t̚ɔ̹ẽɔ̜u̟e̠ël̴n̴ɫe̽e̝ɹ̝m̩n̩l̩e̞β̞e̯e̘e̙ĕe̋éēèȅx͜xx͡x↓↑→↗↘]+$");
 
-        public override ValidationResult Validate(object value, CultureInfo cultureInfo)
+        // Fixup IPA text by replacing spaces
+        private void PhoneticName_TextChanged(object sender, TextChangedEventArgs e)
         {
-            if (value == null)
-            {
-                return ValidationResult.ValidResult;
-            }
-            string val = value.ToString();
-            if (ipaRegex.Match(val).Success)
+            if (sender is TextBox textBox)
             {
-                return ValidationResult.ValidResult;
-            }
-            else
-            {
-                return new ValidationResult(false, "Invalid IPA");
+                if (textBox.IsLoaded)
+                {
+                    // Replace any spaces, maintaining the original caret position
+                    int caretIndex = textBox.CaretIndex;
+                    textBox.Text = textBox.Text.Replace(" ", "ˈ");
+                    textBox.CaretIndex = caretIndex;
```

Comment:
The docs don't say what happens if you set a caret index that is greater than the new text length but we should probably Max() the old index with the new length to be safe. Or implement the full fix above.

Reply:
I can't think of a circumstance where it'd be necessary but I won't oppose a good defensive programming suggestion so I'll add this.